account_asset_management/tests/test_account_asset_management.py

```diff
@@ -186,7 +186,20 @@ def test_01_nonprorata_basic(self):
         self.assertEqual(ict0.value_depreciated, 500)
         self.assertEqual(ict0.value_residual, 1000)
         vehicle0.validate()
-        vehicle0.depreciation_line_ids[1].create_move()
+        created_move_ids = vehicle0.depreciation_line_ids[1].create_move()
+        for move_id in created_move_ids:
+            move = self.env["account.move"].browse(move_id)
+            self.assertEquals(
+                len(
+                    move.line_ids.filtered(
+                        lambda line: line.analytic_account_id ==
+                        self.env.ref("analytic.analytic_administratif") and
+                        line.analytic_tag_ids ==
+                        self.env.ref("analytic.tag_contract")
+                    )
+                ),
+                1,
+            )
```

Comment:
```suggestion
            self.assertTrue(
                move.line_ids.filtered(
                    lambda line: line.analytic_account_id ==
                    self.env.ref("analytic.analytic_administratif") and
                    line.analytic_tag_ids ==
                    self.env.ref("analytic.tag_contract")
                 )
            )
```

Reply:
Thanks for the suggestion @pedrobaeza 
However, this check with the evaluation of the `len()` also asserted that the analytic data was not being propagated to the lines that it shouldn't, but only to the ones of type "expense": https://github.com/OCA/account-financial-tools/pull/1149/files#diff-5318cd3560cdb2d23d2889fe39c5b71745c140810b20f64f0c9aa5b24b8b602bR200-R204
We will lose that with this change... Is that a problem?

Reply:
You are not testing that, but testing that there's only line with such account. If you want to test that, filter by account type, and then `assertEqual` its content.

Reply:
Ok, thanks! I made the change. I now test that the lines with analytic data exist (`assertTrue`) and that they match the appropriate lines, as you suggested (`assertEqual`)